api/src/test/scala/za/co/absa/loginsvc/rest/config/jwt/AwsSecretsManagerKeyConfigTest.scala

```diff
@@ -60,17 +89,104 @@ class AwsSecretsManagerKeyConfigTest extends AnyFlatSpec with Matchers {
   }
 
   it should "fail on non-negative keyPhaseOutTime" in {
-    awsSecretsManagerKeyConfig.copy(keyPhaseOutTime = Option(5.milliseconds)).validate() shouldBe
+    awsSecretsManagerKeyConfig.copy(keyPhaseOutTime = Option(5.milliseconds), keyLayOverTime = None).validate() shouldBe
       ConfigValidationError(ConfigValidationException(s"keyPhaseOutTime must be at least ${KeyConfig.minKeyPhaseOutTime}"))
   }
 
   it should "fail on keyPhaseOutTime being configured without keyRotationTime" in {
-    awsSecretsManagerKeyConfig.copy(pollTime = None).validate() shouldBe
+    awsSecretsManagerKeyConfig.copy(pollTime = None, keyLayOverTime = None).validate() shouldBe
       ConfigValidationError(ConfigValidationException(s"keyPhaseOutTime can only be enable if keyRotationTime is enable!"))
   }
 
+  it should "fail on non-negative keyLayOverTime" in {
+    awsSecretsManagerKeyConfig.copy(keyLayOverTime = Option(5.milliseconds)).validate() shouldBe
+      ConfigValidationError(ConfigValidationException(s"keyLayOverTime must be at least ${KeyConfig.minKeyLayOverTime}"))
+  }
+
+  it should "fail on keyLayOverTime being configured without keyRotationTime" in {
+    awsSecretsManagerKeyConfig.copy(pollTime = None, keyPhaseOutTime = None).validate() shouldBe
+      ConfigValidationError(ConfigValidationException(s"keyLayOverTime can only be enable if keyRotationTime is enable!"))
+  }
+
   it should "fail on missing value" in {
     awsSecretsManagerKeyConfig.copy(secretName = null).validate() shouldBe
       ConfigValidationError(ConfigValidationException("secretName is empty"))
   }
+
+  behavior of "fetchKeySetsFromCloud"
+
+  it should "not use keyLayOver when previousKey is not available" in {
+    when (mockSecretsUtil.fetchSecret(anyString(),anyString(),any[Array[String]](),eqMatch(None)))
+      .thenReturn(currentSecret)
+    when(mockSecretsUtil.fetchSecret(anyString(),anyString(),any[Array[String]](),eqMatch(Some("AWSPREVIOUS"))))
+      .thenReturn(None)
+
+    val keysDuringLayover = awsSecretsManagerKeyConfig.fetchKeySetsFromCloud(mockSecretsUtil)
+
+    assert(keysDuringLayover._1.getPrivate == currentKeyPair.getPrivate)
+    assert(keysDuringLayover._1.getPublic == currentKeyPair.getPublic)
+    assert(keysDuringLayover._2.isEmpty)
```

Comment:
```suggestion
    val (currentKey, optPreviousKey) = awsSecretsManagerKeyConfig.fetchKeySetsFromCloud(mockSecretsUtil)

    assert(currentKey.getPrivate == currentKeyPair.getPrivate)
    assert(currentKey.getPublic == currentKeyPair.getPublic)
    assert(optPreviousKey.isEmpty)
```

This has been already mentioned but reiterating - please try not to use `._1` `._2` unless warrantied -- this is harder to read. Instead, please use named values to better signify their meaning. This makes the code much easily digestable.

(feel free to apply else where)